cai_causal_graph/graph_components.py

```diff
@@ -48,7 +48,7 @@ def __init__(
         self._identifier = identifier
 
         self.variable_type: NodeVariableType = variable_type
-        self.meta = dict() if meta is None else meta
+        self.meta = dict() if meta is None else deepcopy(meta)
```

Comment:
Why do we deepcopy here?

Reply:
Added to align with `TimeSeriesNode` but will remove the copy of meta in both places.